cmd/up.go

```diff
@@ -401,19 +408,82 @@ func (cmd *UpCmd) devPodUpMachine(
 			client.AgentPath(),
 			client.AgentURL(),
 			true,
-			command,
+			sshCmd,
 			stdinReader,
 			stdoutWriter,
 			writer,
 			log.ErrorStreamOnly(),
 		)
-		if err != nil {
+		if err != nil && !errors.Is(err, context.Canceled) && !strings.Contains(err.Error(), "signal: ") {
 			errChan <- fmt.Errorf("executing agent command: %w", err)
 		} else {
 			errChan <- nil
 		}
 	}()
 
+	// create pipes
+	stdoutReader2, stdoutWriter2, err := os.Pipe()
+	if err != nil {
+		return nil, err
+	}
+	stdinReader2, stdinWriter2, err := os.Pipe()
+	if err != nil {
+		return nil, err
+	}
+	defer stdoutWriter2.Close()
+	defer stdinWriter2.Close()
+
+	// connect to container
+	go func() {
+		defer cancel()
+
+		log.Debugf("Attempting to create SSH client")
+		// start ssh client as root / default user
+		sshClient, err := devssh.StdioClient(stdoutReader, stdinWriter, false)
+		if err != nil {
+			errChan <- errors.Wrap(err, "create ssh client")
+			return
+		}
+		defer log.Debugf("Connection to SSH Server closed")
+		defer sshClient.Close()
+
+		log.Debugf("SSH client created")
+
+		sess, err := sshClient.NewSession()
+		if err != nil {
+			errChan <- errors.Wrap(err, "create ssh session")
+		}
+		defer sess.Close()
+
+		log.Debugf("SSH session created")
+
+		var identityAgent string
+		if identityAgent == "" {
+			identityAgent = os.Getenv("SSH_AUTH_SOCK")
+		}
+
+		if identityAgent != "" {
+			err = gosshagent.ForwardToRemote(sshClient, identityAgent)
+			if err != nil {
+				errChan <- errors.Wrap(err, "forward agent")
+			}
+			err = gosshagent.RequestAgentForwarding(sess)
```

Comment:
just leaving this as a marker: @neogopher and me had a discussion around why this works on the session even though we create a new one in `devssh.Run`, he investigated a potential bug in the ssh library dependency and I'll also check it out this week to be sure